src/main/java/de/blackforestsolutions/dravelopsdatamodel/Leg.java

```diff
@@ -1,49 +1,55 @@
 package de.blackforestsolutions.dravelopsdatamodel;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.hazelcast.nio.serialization.Portable;
+import com.hazelcast.nio.serialization.PortableReader;
+import com.hazelcast.nio.serialization.PortableWriter;
 import lombok.AccessLevel;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 import lombok.experimental.Accessors;
-import lombok.extern.slf4j.Slf4j;
 import org.springframework.data.geo.Distance;
-import org.springframework.data.geo.Point;
+import org.springframework.data.geo.Metrics;
 
+import java.io.IOException;
+import java.io.Serializable;
 import java.time.Duration;
 import java.util.LinkedList;
-import java.util.UUID;
+
+import static de.blackforestsolutions.dravelopsdatamodel.hazelcast.DravelOpsPortableFactory.DRAVEL_OPS_FACTORY_ID;
+import static de.blackforestsolutions.dravelopsdatamodel.hazelcast.DravelOpsPortableFactory.LEG_CLASS_ID;
+import static de.blackforestsolutions.dravelopsdatamodel.hazelcast.classdefinition.LegClassDefinition.*;
 
 @Getter
-@Slf4j
 @JsonDeserialize(builder = Leg.LegBuilder.class)
-public final class Leg {
+public final class Leg implements Serializable, Portable {
 
-    private final UUID id;
+    private static final long serialVersionUID = 5393486245718564673L;
 
-    private final TravelPoint departure;
+    private TravelPoint departure;
 
-    private final TravelPoint arrival;
+    private TravelPoint arrival;
 
-    private final Duration delayInMinutes;
+    private Duration delayInMinutes;
 
-    private final Distance distanceInKilometers;
+    private Distance distanceInKilometers;
 
-    private final VehicleType vehicleType;
+    private VehicleType vehicleType;
 
-    private final LinkedList<Point> waypoints;
+    private LinkedList<Point> waypoints;
 
-    private final TravelProvider travelProvider;
+    private TravelProvider travelProvider;
 
-    private final String vehicleNumber;
+    private String vehicleNumber;
 
-    private final String vehicleName;
+    private String vehicleName;
 
-    private final LinkedList<TravelPoint> intermediateStops;
+    private LinkedList<TravelPoint> intermediateStops;
```

Comment:
Maybe we need this to implement Portable, but we arent immutable anymore in this case. How do we enforce to not use setter for this method?

Reply:
Wir sollten meiner Meinung nach Portable wieder entfernen. Ich habe noch keine Moeglichkeit gefunden Portable in der Server-Client-Arhcitektur zu konfigurieren.

Reply:
Done